torch/csrc/autograd/functions/special.cpp

```diff
@@ -231,22 +235,21 @@ bool Eval::replaceSubgraph(const variable_list& inputs, const variable_list& _ou
 
     // Replace begins with pointers to output nodes.
     // This detaches the subgraph from the full backward graph.
-    for (auto & begin : subgraph.boundary.begins) {
-      auto & fn = begin.first;
-      auto offset = begin.second;
-      fn->next_functions[offset] = std::make_pair(ends_to_outputs.at(fn->next_functions[offset]), 0);
+    for (auto& begin : subgraph.boundary.begins) {
+      begin.function->next_functions[begin.port] = FunctionPort(
+          ends_to_outputs.at(begin.function->next_functions[begin.port]));
```

Comment:
Um how does it even compile? It shouldn't be possible to make a `FunctionPort` without specifying `input_nr`. It doesn't look equivalent to what was there before

Reply:
The `input_nr` (currently `port`) defaults to zero in the constructor.